web/src/pages.rs

```diff
@@ -0,0 +1,30 @@
+use axum::response::{IntoResponse, Response};
+use db::users::User;
+
+use crate::templates::{Template, TemplatedPage};
+
+pub async fn home(t: Template, user: Option<User>) -> Response {
+    match user {
+        Some(user) => user_dashboard(t, user).await.into_response(),
+        None => landing(t).await.into_response(),
+    }
+}
+
+pub async fn landing(t: Template) -> TemplatedPage {
+    t.render(maud::html! {
+        h1 { "Knowledge" }
+        h2 { "A cool app that needs a new name" }
+
+        p { "Welcome to Knowledge! Please log in." }
+        a href="/login" { "Login" }
+    })
+}
+
+pub async fn user_dashboard(t: Template, user: User) -> TemplatedPage {
```

Comment:
You can also add `user: User` without the wrapping option to your route handler!

And this time if the user is NOT logged in they will not be able to go to this route, and will instead be redirects to the `/login` page!